wire-runtime/src/main/java/com/squareup/wire/Message.java

```diff
@@ -160,46 +161,40 @@ protected void setBuilder(Builder builder) {
     return adapter.fromInt(value);
   }
 
+  /**
+   * Create a {@code byte[]} of the serialized form of the data.
+   * <p>
+   * Note: This is an inefficient means of serializing data. For best performance, call
+   * {@link Wire#adapter} and then {@link MessageAdapter2#write}.
+   */
   @SuppressWarnings("unchecked")
   public byte[] toByteArray() {
```

Comment:
I'd love to remove this method too. You can't serialize a JSON type by calling `toJson()` on it. This seems like a holdover from trying to be protobuf which we can't do based on the fundamental choices herein. It's a convenience method to inefficiency.

But it requires a lot of test changes, so... followup, if so!